### DISCUSSION TITLE
Why create a new game engine instead of using established ones?

### DISCUSSION THREAD

**Discussion by davidak** [matrix-org/thirdroom#162] [General]
I really like the idea of having a Metaverse that is not the dystopian future corporations like Facebook are building. So i'm very interested in this project.

I would like to know why you don't use an established game engine like Godot, Unity or Unreal.

You probably have considered that, so why did you choose not to use them?

**Comment by robertlong** [comment]
Hey @davidak 👋 We're using [Three.js](https://threejs.org/) for our rendering engine and a number of other open source libraries to bootstrap our engine. The reason for doing this is that Godot, Unity, and Unreal have traditionally not been very friendly for WebGL export and building an open platform on top of closed engines such as Unity and Unreal would likely be a mistake in the long term. The bundle sizes of apps exported from Godot, Unity, and Unreal are pretty large and can take a while to download. We're optimizing for the strengths of the Web and with it we need to design around the constraints the Web has. It's certainly a challenge but one we think will pay off immensely and make way for a much more open platform for creators and users alike.

**Reply by davidak** [reply]
I see.

You are probably aware of [WebGPU](https://en.wikipedia.org/wiki/WebGPU) and [WebXR](https://en.wikipedia.org/wiki/WebXR)?

**Reply by ara4n** [reply]
yes, we are.

**Comment by lanmower** [comment]
Man this is exactly why I'm excited about thirdroom, its possible to optimize the hell out of content nowadays with draco and ktx2, its nice to see an environment that can be well aligned with those workflows
